### PR TITLE
Add autoscroll option to deployment log

### PR DESCRIPTION
**Pull request by jeremy-moffitt**
add a checkbox to autoscroll the deployment log.
addresses https://github.com/SUSE-Enceladus/blue-horizon/issues/67

### CODE REVIEW

**Comment by jeremy-moffitt** [comment]
I cant tell if there's a spacing issue in deploys.js or not, the various editors I tried are interpreting tabs differently (all of the new lines use spaces instead of tabs)

**Comment by jeremy-moffitt** [comment]
auto_scroll
with and without autoscroll with some nonsense terraform output

File: app/assets/javascripts/deploys.js [hunk 1]
```diff
@@ -55,6 +55,10 @@ function fetch_output(finished, intervalId) {
         clearTimeout(intervalId);
       } else {
 	$(".pre-scrollable").html(data.new_html);
+        let autoscroll = $('#deploy_log_autoscroll').prop('checked');
+        if(autoscroll){
```
Comment (jesusbv) [inline app/assets/javascripts/deploys.js]: [Space before blocks rule](https://standardjs.com/rules-en.html)
image

File: app/assets/javascripts/deploys.js [hunk 2]
```diff
@@ -55,6 +55,10 @@ function fetch_output(finished, intervalId) {
         clearTimeout(intervalId);
       } else {
 	$(".pre-scrollable").html(data.new_html);
+        let autoscroll = $('#deploy_log_autoscroll').prop('checked');
```
Comment (jesusbv) [inline app/assets/javascripts/deploys.js]: it should use **const** instead of **let** as `autoscroll` is not going to be reassigned.

**Comment by jesusbv** [comment]
Thank you for the contribution, Jeremy.
Other than the changes in JavaScript syntax, looks good.

I'm not a UX expert so rather question if there is a better place for the checkbox,
I'm not sure what's the best place for it: at the bottom right (of the output box), at the top right or stay at the bottom left.

Thoughts?

**Comment by jeremy-moffitt** [comment]
Thanks for the feedback, I'll see if I can loop in someone from the UX team on the question of placement, and will fix the issues you identified, probably wont be for a day or two, but this is clearly not urgent.

**Comment by jeremy-moffitt** [comment]
The guidance from Sorin on position of the checkbox is as follows:
```For this specific case, I cannot think in any standard for it, usually, it depends on other elements next to the checkbox.
By checking the screenshots I saw that the navigation it is placed on the right side, to keep it more consistent I would say that you could also move the checkbox to the right but, moving it right on top of the navigation it might get lost or confused.

I  think that top right would fit the best (since it’s more close to the context, scrollbar, and it won’t interfere with the navigation.)```

so I'll move it to the top right

**Comment by jesusbv** [comment]
> > I  think that top right would fit the best (since it’s more close to the context, scrollbar, and it won’t interfere with the navigation.)
> 
> so I'll move it to the top right

Thank you for the effort. At the top right was my preference but again, not an expert, so good it's clear now.

**Comment by jeremy-moffitt** [comment]
auto_scroll_floatright
updated screenshot... I think I tackled the other issues, but if I missed something or if there's any other changes needed, please let me know

**Pull request merged**